src/OptimizeImageTrait.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the ekino/tiny-png-sonata-media-bundle project.
+ *
+ * (c) Ekino
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Ekino\TinyPngSonataMediaBundle;
+
+use Ekino\TinyPngSonataMediaBundle\Consumer\OptimizeImageConsumer;
+use Sonata\NotificationBundle\Backend\BackendInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+/**
+ * Trait OptimizeImageTrait.
+ *
+ * @author William JEHANNE <william.jehanne@ekino.com>
+ */
+trait OptimizeImageTrait
+{
+    /**
+     * @var BackendInterface
+     */
+    public $backend;
+
+    /**
+     * @var SessionInterface
+     */
+    public $session;
+
+    /**
+     * @param Request|null $request
+     *
+     * @return mixed
+     */
+    public function optimizeAction(Request $request = null)
+    {
+        $this->backend->createAndPublish(OptimizeImageConsumer::CONSUMER_TYPE, [
+            'mediaId' => $request->get('id'),
+        ]);
+
+        $this->session->getFlashBag()->add(
+            'sonata_flash_success',
+            'Successfully launch the optimization of your media.'
+        );
+
+        return $this->redirectToList();
+    }
+
+    /**
+     * @param BackendInterface $backend
+     */
+    public function setBackend(BackendInterface $backend)
```

Comment:
If the properties are public, having setters seems useless.

Reply:
They should not be :)